v2/contribute/decisions/multitenancy/0012-how-to-pass-tenant-id-to-backend.mdx

```diff
@@ -0,0 +1,48 @@
+---
+id: "0012"
+title: How to pass tenant id to backend?
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader"
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro"
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon"
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut"
+
+# How to pass tenant id to backend?
+
+<DecisionHeader status="proposed" lastUpdate="2022-12-22" created="2022-12-22" deciders={["rishabhpoddar", "sattvikc"]} proposedBy={["sattvikc"]} />
+
+## Context and Problem Statement
+
+Tenant Id is configured in the frontend using `async getTenantId` config function.
+
+What is the best way to pass the tenant Id to the backend auth APIs?
+
+## Considered Options
+
+* Pass it as API query parameter or API request body
+* Pass it as API header
+
+## Decision Outcome
+
+Chosen option: Pass it as API query parameter or API request body, because
+
+* Easy to use and understand
+* Passing as header has more cons than this approach
+
+
+## Pros and Cons of the Options
+
+### Pass it as API query parameter or API request body
+
+Further details if necessary
+
+<ArgumentPro> Easy to use and understand </ArgumentPro>
+<ArgumentCon> For single tenant users, an extra input </ArgumentCon>
+
```

Comment:
what about the whole discussion we had about the backend also having a function to get the tenantId and the input to that is the frontend tenant ID along with the userContext (which contains the request object)?

Why did you miss that whole thing?